dips/dip-158.md

```diff
@@ -0,0 +1,794 @@
+    "status": {
+      "status": "abort",
+      "abort_code": "void",
+      "abort_message": "abort message"
+    }
+  }
+}
+
+
+
+
+```
+#### VoidCommand Response
+Field|Type|Required?|Description|
+|-|-|-|-|
+|status|StatusObject|Y|Sender status. Used to verify that the sender VASP acknoledges the void command by responding with the abort status|
+|
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a",
+  "result": {
+    "_ObjectType": "StatusObject",
+    "status": {
+      "status": "abort"
+    }
+  }
+}
+```
+
+### AbortCommand
+
+| Functionality |Direction| Description | Command Type | Request Data | Response Data |
+|-|-|-|-|-|-|
+|ABORT_PAYMENT|Sender > Receiver / Receiver > Sender|This functionality allows both parties to abort the payment. For example, the receiver may want to abort the payment following failed risk checks|AbortCommand|Reference ID ; Requester (Sender / Receiver) Status|Responder (Sender / Receiver) Status|
+|
+
+#### AbortCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|sender / receiver|PaymentActorObject|Y|Indicates the initiator of the abort command (can be either the sender or the receiver). Must contain a nested StatusObject (other elements are not required)|
+|
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "AbortCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "AbortCommand",    
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+    "sender": {
+      "status": {
+        "status":"abort",
+        "abort_code": "rejected",
+        "abort_message": "abort message"
+      }
+    }
+  }
+}
+
+```
+#### AbortCommand Response
+Field|Type|Required?|Description|
+|-|-|-|-|
+|status|StatusObject|Y|Responder status. Used to verify that the responding VASP acknoledges the abort command by responding with the abort status|
+|
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a",
+  "result": {
+    "_ObjectType": "StatusObject",
+    "status": {
+      "status": "abort"      
+    }
+  }
+}
+
+```
+
+---
+## Payment Sequences
+---
+
+This section describes the expected sequences in P2M scenarios.
+Each scenario is expressed using a sequence of Sender / Receiver functionality.
+
+**Reminder:** The step of exchanging a common payment identifier has been fulfilled
+
+### Charge Sequence
+
+| Step | Functionality | Command Type | Triggered by | Description |
+|-|-|-|-|-|
+|1|GET_INFO|InfoCommand|Sender|The wallet requests payment details using the reference id provided (i.e. the common payment identifier). The receiver will respond with the payment details and merchant data. The wallet will display the payment details so the customer can approve or reject|
```

Comment:
it is a bit cumbersome to use repeatedly "the ref id provided (i.e., ...)" .

can we name it above "The Common Reference ID", the first time we introduce the requirement that pre-flight exchange of payment identifier/tx referenceID , and then use it consistently?

Reply:
Given that I added an item to the glossary I will remove the part in parenthesis.
Less is more in this case :)